File: Stage30/Python/Problem3273.py
import sys
input = sys.stdin.readline

def main():
    n = int(input())
    arr = list(map(int, input().split()))
    x = int(input())

    arr.sort()

    start = 0
    end = n - 1
    count = 0

    while start < end:
        s = arr[start] + arr[end]

        if s == x:
            count += 1
            start += 1
            end -= 1
        elif s < x:
            start += 1
        else:
            end -= 1

    print(count)

if __name__ == "__main__":
    main()

File: Stage30/C/Problem3273.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int main(void) {
    int n;
    scanf("%d", &n);

    int arr[100000];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int x;
    scanf("%d", &x);

    qsort(arr, n, sizeof(int), compare);

    int start = 0;
    int end = n - 1;
    int count = 0;

    while (start < end) {
        int sum = arr[start] + arr[end];

        if (sum == x) {
            count++;
            start++;
            end--;
        } else if (sum < x) {
            start++;
        } else {
            end--;
        }
    }

    printf("%d\n", count);

    return 0;
}